Can't build on alpine linux

The used go-crypt module contains glibc specific code. Alpine linux is using musl instead of glibc.

```
owncloud/idm # make
running gofmt ...
retrieving dependencies ...
# github.com/Azure/go-ntlmssp v0.0.0-20211209120228-48547f28849e
...
building cmd/idmd ...
CGO_ENABLED=1 go build \
        -mod=vendor \
        -trimpath \
        -tags release \
        -buildmode=exe \
        -ldflags '-s -w -buildid=reproducible/0.3.0-41-gb89b052 -X github.com/libregraph/idm/version.Version=0.3.0-41-gb89b052 -X github.com/libregraph/idm/version.BuildDate=2022-02-17T17:28:53Z -extldflags -static' \
        -o bin/idmd ./cmd/idmd
# github.com/amoghe/go-crypt
vendor/github.com/amoghe/go-crypt/version.go:6:10: fatal error: gnu/libc-version.h: No such file or directory
    6 | #include <gnu/libc-version.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:47: cmd/idmd] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -34,6 +34,7 @@ PKGS     = $(or $(PKG),$(shell $(GO) list -mod=readonly ./... | grep -v "^$(PACK
 TESTPKGS = $(shell $(GO) list -mod=readonly -f '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' $(PKGS) 2>/dev/null)
 CMDS     = $(or $(CMD),$(addprefix cmd/,$(notdir $(shell find "$(PWD)/cmd/" -maxdepth 1 -type d))))
 TIMEOUT  = 30
+BUILD_TAGS ?=
 
 # Build
 
@@ -47,7 +48,7 @@ $(CMDS): vendor ; $(info building $@ ...) @
 	CGO_ENABLED=$(CGO_ENABLED) $(GO) build \
 		-mod=vendor \
 		-trimpath \
-		-tags release \
+		-tags "release $(BUILD_TAGS)" \
 		-buildmode=exe \
 		-ldflags '-s -w -buildid=reproducible/$(VERSION) -X $(PACKAGE)/version.Version=$(VERSION) -X $(PACKAGE)/version.BuildDate=$(DATE) -extldflags -static' \
 		-o bin/$(notdir $@) ./$@

--- a/pkg/ldappassword/crypt.go
+++ b/pkg/ldappassword/crypt.go
@@ -1,4 +1,4 @@
-//go:build linux || freebsd || netbsd
+//go:build !disable_crypt && (linux || freebsd || netbsd)
 
 /*
  * SPDX-License-Identifier: Apache-2.0

--- a/pkg/ldappassword/crypt_disabled.go
+++ b/pkg/ldappassword/crypt_disabled.go
@@ -1,3 +1,5 @@
+//go:build disable_crypt || darwin || windows
+
 /*
  * SPDX-License-Identifier: Apache-2.0
  * Copyright 2021 The LibreGraph Authors.

